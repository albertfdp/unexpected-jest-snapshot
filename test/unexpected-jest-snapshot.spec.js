const unexpected = require('unexpected')
const unexpectedJestSnapshot = require('../src')

const React = require('react')
const TestRenderer = require('react-test-renderer')

const expect = unexpected.clone()
  .use(unexpectedJestSnapshot)

const Component = ({ name }) => (
  React.createElement('div', null, `Name: ${name}`)
)

describe('unexpected-jest-snapshot', () => {
  describe('snapshots', () => {
    it('creates a snapshot', () => {
      expect('foo', 'to match snapshot')
    })
  })

  describe('with react-test-renderer', () => {
    it('creates a snapshot of the rendered component', () => {
      expect(
        React.createElement(Component, { name: 'foo' }),
        'to match snapshot'
      )
    })
  })
})