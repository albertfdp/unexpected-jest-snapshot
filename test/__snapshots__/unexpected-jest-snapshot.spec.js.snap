exports[`unexpected-jest-snapshot snapshots creates a snapshot 1`] = `"foo"`;

exports[`unexpected-jest-snapshot with react-test-renderer creates a snapshot of the rendered component 1`] = `
<div>
  Name: foo
</div>
`;
